plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java

```diff
@@ -207,13 +209,29 @@ private String requestCompletion(
                             if (e instanceof HttpException) {
                                 RuntimeUtils.pause(1000);
                             } else {
-                                responseSize /= 2;
+                                // Extracts resulted prompt size from the error message and resizes max response to
+                                // value lower that (maxTokens - prompt size)
+                                Matcher matcher = Pattern.compile("([0-9]+)").matcher(e.getMessage());
+                                ArrayList<Integer> numbers = new ArrayList<>();
+                                while (matcher.find()) {
+                                    String numberStr = matcher.group();
+                                    int number = Integer.parseInt(numberStr);
+                                    numbers.add(number);
+                                }
+                                if (numbers.size() < 4) {
+                                    throw e;
+                                }
+                                Integer promptSize = numbers.get(2);
+                                responseSize = Math.min(responseSize,
+                                    GPTModel.getByName(getModelName()).getMaxTokens() - promptSize - 1);
```

Comment:
Ugh, can we rely on some words in the message to get the required size instead of relying on the ordinal position of the number token in this message?